implementations/rust/ockam/ockam_api/src/nodes/service/forwarder.rs

```diff
@@ -49,14 +94,135 @@ impl NodeManager {
                     remote_address = %b.remote_address(),
                     "CreateForwarder request processed, sending back response"
                 );
-                Ok(Response::ok(req.id()).body(b).to_vec()?)
+                Ok(Response::ok(rid).body(b).to_vec()?)
             }
             Err(err) => {
                 error!(?err, "Failed to create forwarder");
-                Ok(Response::builder(req.id(), Status::InternalServerError)
+                Ok(Response::builder(rid, Status::InternalServerError)
                     .body(err.to_string())
                     .to_vec()?)
             }
         }
     }
 }
+
+/// Get address prefix corresponding to the API host (if any).
+#[rustfmt::skip]
+fn api_host(input: &MultiAddr) -> Option<MultiAddr> {
+    let mut protos = input.iter();
+    if !matches!(protos.next().map(|p| p.code()), Some(DnsAddr::CODE | Ip4::CODE | Ip6::CODE)) {
+        return None;
+    }
+    if !matches!(protos.next().map(|p| p.code()), Some(Tcp::CODE)) {
+        return None;
+    }
+    if let Some(p) = protos.next() {
+        if let Some(p) = p.cast::<Service>() {
+            if &*p == "api" {
+                return MultiAddr::default().try_with(input.iter().take(3)).ok();
+            }
+        }
+    }
+    None
+}
+
+/// Create a secure channel to an API node or return the address route as is.
+async fn connect(
```

Comment:
Should it be a NodeManager's function?

Reply:
I deliberately made it a private function of this module for the time being so that other parts of the `NodeManager` would not be tempted to use it.

Reply:
But it can be private even if it's a NodeManager's function

Reply:
Not sure I get what you mean. If it is a private *method* of the `NodeManager` it would be available in the whole scope of it vs. just being a private function of this module. Did you have something else in mind?

Reply:
AFAIK private function of the struct, that is defined in submodule, is only available from that submodule. In other words this `connect` function would be available from this file only doesn't matter if it's a static function of a struct's function

Reply:
Thanks, I did not think it would behave like this but it makes sense.